### PR TITLE
feat: add release-binaries workflow

### PR DESCRIPTION
**Pull request by rvagg**
Closes: https://github.com/ipld/go-car/issues/354

### CODE REVIEW

File: .github/workflows/release-binaries.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - uses: wangyoucao577/go-release-action@v1.34
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        goos: ${{ matrix.goos }}
+        goarch: ${{ matrix.goarch }}
+        goversion: "https://go.dev/dl/go1.19.5.linux-amd64.tar.gz"
+        project_path: "./cmd/lassie"
+        binary_name: "lassie"
```
Comment (willscott) [inline .github/workflows/release-binaries.yml:28]: these change?

**Review by masih — COMMENTED** [review]
Alternatively you can use goreleaser like the one i have [set up for `index-provider` CLI](https://github.com/ipni/index-provider/blob/main/.github/workflows/releaser-bin.yml). That also uploads binaries to github releases amongst other stuff; [example](https://github.com/ipni/index-provider/releases/tag/v0.10.1).

**Comment by rvagg** [comment]
is goreleaser superior to this setup? I'm not really fussed as long as it works, I'd love to have consistency across projects but it seems like everyone's having a go at something different with these

**Comment by masih** [comment]
I too am not fussed. For me go releaser does more with less config; that's all. Feel free to merge this as is.

**Comment by rvagg** [comment]
fair enough, it is a simpler setup and the name "goreleaser" makes it feel more official...
have changed the config, ptal

File: .github/workflows/release-binaries.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: GoReleaser
```
Comment (masih) [inline .github/workflows/release-binaries.yml:1]: For goreleaser binary build to do the right thing in this repo a config like [this](https://github.com/ipni/index-provider/blob/main/.goreleaser.yaml) is needed. Once that's added try running `goreleaser check` to verify the config, and give `goreleaser --rm-dist --snapshot` a shot to make sure binaries are built as you expect them to (written in `dist` directory by default).

I also recommend adding `dist` to `.gitingore`.

**Review by masih — CHANGES_REQUESTED** [review]

File: .github/workflows/release-binaries.yml (hunk 2 above)
Comment (rvagg) [reply .github/workflows/release-binaries.yml:1]: > does more with less config
he said!

ok, added the config and tested it all out

Comment (masih) [reply .github/workflows/release-binaries.yml:1]: lol

**Review by masih — APPROVED** [review]

**Pull request merged**